src/test/java/me/karakelley/http/PublicDirectoryTest.java

```diff
@@ -0,0 +1,18 @@
+package me.karakelley.http;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class PublicDirectoryTest {
+
+  @Test
+  void listFiles() {
+    PublicDirectory directory = new PublicDirectory("./src/test/resources/directory/");
```

Comment:
Hard-coding paths in tests is a smell. You're probably dependent on your machine, or how the code is run (intellij vs compiled class files vs jars, etc).

It would be better to create a temp directory in your test, add some files to it, and then delete the temp directory once you're done. The suggestion above about the logging appender could also apply here...

```
// no directory exists
withTempDirectory(dir -> {
  // add a couple files
  // make your assertions
});
// no directory exists
```